src/agent/command_handler/include/command_handler.hpp

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <boost/asio.hpp>
+#include <command_store.hpp>
+
+#include <iostream>
+#include <optional>
+#include <queue>
+
+namespace command_handler
+{
+    class CommandHandler
+    {
+    public:
+        CommandHandler();
+
+        template<typename T>
+        boost::asio::awaitable<void>
+        ProcessCommandsFromQueue(const std::function<std::optional<T>()> GetCommandFromQueue,
+                                 const std::function<bool()> PopCommandFromQueue,
+                                 const std::function<int(T&)> DispatchCommand)
+        {
+            using namespace std::chrono_literals;
+            const auto executor = co_await boost::asio::this_coro::executor;
+            std::unique_ptr<boost::asio::steady_timer> expTimer = std::make_unique<boost::asio::steady_timer>(executor);
+
+            while (true)
+            {
+                auto cmd = GetCommandFromQueue();
+                if (cmd == std::nullopt)
+                {
+                    expTimer->expires_after(1000ms);
+                    co_await expTimer->async_wait(boost::asio::use_awaitable);
+                    std::cout << "Queue is empty - WAITING" << "\n";
+                    continue;
+                }
+
+                m_commandStore.StoreCommand(cmd.value());
+                PopCommandFromQueue();
+                DispatchCommand(cmd.value());
+
+                cmd.value().m_status = command_store::Status::SUCCESS;
+                cmd.value().m_result = "Successfully executed";
```

Comment:
These values should be obtained from the result of `DispatchCommand` execution.

Reply:
Done